### PR TITLE
PST-004-connect-with-todos-api

### PR DESCRIPTION
**Pull request by skrzypczyk-pawel**

### CODE REVIEW

File: src/components/Todo/Todo.tsx [hunk 1]
```diff
@@ -22,20 +21,25 @@ export const Todo: FC<Props> = ({ todo }) => {
       role="menuitem"
       tabIndex={Math.random()}
     >
-      <p className={css(styles.id, typography.altBody1)}>{todo.id}</p>
+      <div className={css(styles.id, typography.altBody1)}>
```
Comment (skrzypczyk-pawel) [inline src/components/Todo/Todo.tsx:24]: I tried to find a way to set paragraph like a div, but I didn't find any good way to do that.

File: src/components/Todo/TodoList.tsx [hunk 2]
```diff
@@ -1,14 +1,29 @@
-import React, { FC } from "react";
+import React, { FC, useEffect, useState } from "react";
 
 import { Todo } from "src/components/Todo/Todo";
-import { todos } from "src/constants";
+import Loader from "src/components/Loader";
+import { ITodo } from "src/types";
 
 export const TodoList: FC = () => {
+  const [tasks, setTasks] = useState([]);
```
Comment (Stared55) [inline src/components/Todo/TodoList.tsx]: add type

File: src/components/Todo/TodoList.tsx [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+
+  useEffect(() => {
+    fetch("https://jsonplaceholder.typicode.com/todos")
```
Comment (Stared55) [inline src/components/Todo/TodoList.tsx]: remove to a new function (fetchTodos) and handle loading state

File: src/components/Todo/TodoList.tsx (hunk 2 above)
Comment (Stared55) [inline src/components/Todo/TodoList.tsx]: add new state: `const [loading, setLoading] = useState<boolean>(false);`

File: src/components/Todo/TodoList.tsx (hunk 3 above)
Comment (Stared55) [reply src/components/Todo/TodoList.tsx]: and this time replace then.catch with async/await

File: src/types/todos.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ export interface ITodo {
   id: string;
   title: string;
   category?: Category;
+  completed?: boolean;
```
Comment (Stared55) [inline src/types/todos.ts]: What about our `status` property ?

File: src/components/Todo/TodoList.tsx [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
-    </div>
+    <>
+      {!!tasks.length ? (
+        tasks.map((todo) => {
+          return <Todo todo={todo} />;
+        })
+      ) : (
+        <Loader />
+      )}
+    </>
```
Comment (Stared55) [inline src/components/Todo/TodoList.tsx:44]: loading ? `<Loader />`  : {!!tasks.length ? (
        tasks.map((todo) => {
          return <Todo todo={todo} />;
        })
      )

File: src/components/Todo/TodoList.tsx (hunk 3 above)
Comment (Stared55) [reply src/components/Todo/TodoList.tsx]: `const x = "https://jsonplaceholder.typicode.com/todos"`

File: src/components/Todo/TodoList.tsx [hunk 6]
```diff
@@ -3,26 +3,71 @@ import React, { FC, useEffect, useState } from "react";
...
+  const fetchTodos = async () => {
+    try {
+      setLoading(true);
+      const response = await fetch(todosDataURL);
+      const todosTasks = await response.json();
+      setTasks(todosTasks.filter((task: ITodo) => +task.id <= 30));
+    } catch {
+      alert(i18n.t("todo:alert"));
+    } finally {
+      setLoading(false);
```
Comment (Stared55) [inline src/components/Todo/TodoList.tsx:33]: nice

File: src/components/Todo/TodoList.tsx [hunk 7]
```diff
@@ -3,26 +3,71 @@ import React, { FC, useEffect, useState } from "react";
...
+            todo.description = i18n.t("todo:description2");
+            todo.createdAt = "work";
+          }
+          if (randomNumber() === 2) {
+            todo.priority = "medium";
+            todo.category = "homework";
+            todo.deadline = "01-01-2034";
+            todo.description = i18n.t("todo:description3");
+            todo.createdAt = "house";
+          }
```
Comment (Stared55) [inline src/components/Todo/TodoList.tsx]: Ok, this is very ugly. Today we will do smth together with this shit

**Review by Stared55 — APPROVED** [review]

**Pull request merged**